arches/app/signals.py

```diff
@@ -284,3 +284,49 @@ def ensure_single_default_searchview(sender, instance, **kwargs):
             raise ValidationError(
                 "Only one search logic component can be default at a time."
             )
+
+
+@receiver(post_save, sender=models.Card)
+@receiver(post_delete, sender=models.Card)
+@receiver(post_save, sender=models.CardModel)
+@receiver(post_delete, sender=models.CardModel)
+@receiver(post_save, sender=models.Node)
+@receiver(post_delete, sender=models.Node)
+@receiver(post_save, sender=models.Edge)
+@receiver(post_delete, sender=models.Edge)
+@receiver(post_save, sender=models.FunctionXGraph)
+@receiver(post_delete, sender=models.FunctionXGraph)
+@receiver(post_save, sender=models.GraphXPublishedGraph)
+@receiver(post_delete, sender=models.GraphXPublishedGraph)
+def set_related_graph_has_unpublished_changes_to_true(sender, instance, **kwargs):
+    models.GraphModel.objects.filter(pk=instance.graph_id).update(
+        has_unpublished_changes=True
+    )
+
+
+@receiver(post_save, sender=models.NodeGroup)
+@receiver(post_delete, sender=models.NodeGroup)
+def set_related_graph_has_unpublished_changes_to_true(sender, instance, **kwargs):
+    # NodeGroups have no direct relation to the GraphModel objects,
+    # so this signal can fail to find the node when deleting a Graphs
+    try:
+        node = models.Node.objects.filter(pk=instance.pk).first()
```

Comment:
In case we already have it:
```suggestion
        node = instance.grouping_node
```